### PR TITLE
Merge branch-notes to master

### LINKED ISSUE #10: Add notes
As a student, I want to be able to add notes taken during lecture so that I do no need to manage it physically myself.

### LINKED ISSUE #11: Delete notes
As a student, I want to be able to delete my notes so that I can re-organize information taken down by me.

### LINKED ISSUE #13: View stored notes
As a student, I want to be able to view the notes taken during lecture so that I can refer to them easily.

### PR DESCRIPTION
**Pull request by Woolicious98**
Add notes add, delete and view functions.

Closes #10 
Closes #11 
Closes #13

### CODE REVIEW

File: src/main/java/terminus/command/note/AddCommand.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Override
+    public void parseArguments(String arguments) throws InvalidArgumentException {
+        // Perform required checks with regex
+        if(arguments == null || arguments.isBlank()){
+            throw new InvalidArgumentException("Error: Missing arguments.");
+        }
+        ArrayList<String> argArray = findArguments(arguments);
+        if(!isValidArguments(argArray)){
+            throw new InvalidArgumentException("Error: Missing arguments.");
```
Comment (kelvneo) [inline src/main/java/terminus/command/note/AddCommand.java:46]: Magic literals

File: src/main/java/terminus/command/note/AddCommand.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import terminus.exception.InvalidCommandException;
+import terminus.module.NusModule;
+import terminus.ui.Ui;
+
+public class AddCommand extends Command {
+
+    private String name;
+    private String data;
+
+    private int totalArg;
```
Comment (kelvneo) [inline src/main/java/terminus/command/note/AddCommand.java:22]: Can this be a static final variable?

File: src/main/java/terminus/content/ContentManager.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package terminus.content;
+
+import java.util.ArrayList;
+import terminus.exception.InvalidArgumentException;
+
+public class ContentManager {
+
+    final private static String DATA_SEPARATOR = "";
```
Comment (kelvneo) [inline src/main/java/terminus/content/ContentManager.java:8]: Redundant variable?

File: src/main/java/terminus/common/CommonFormat.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+public class CommonFormat {
+    final public static String COMMAND_NOTE = "note";
+    final public static String COMMAND_ADD = "add";
+    final public static String COMMAND_DELETE = "delete";
+    final public static String COMMAND_VIEW = "view";
+    final public static String COMMAND_BACK = "back";
+    final public static String COMMAND_EXIT = "exit";
+
+    final public static String COMMAND_DELETE_FORMAT = COMMAND_DELETE + " {item number}";
+    final public static String COMMAND_VIEW_FORMAT = COMMAND_VIEW + " {[optional]item number}";
```
Comment (3m0W33D) [inline src/main/java/terminus/common/CommonFormat.java:12]: Regarding the format how should we choose to format this?
For mandatory arguments - `<args>`
For optional - `{args}`
?

**Review by kelvneo — CHANGES_REQUESTED** [review]
Nice job on the entire suite of commands, but needs JUnit testing and styling fixes.

File: src/main/java/terminus/common/Messages.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package terminus.common;
+
+public class Messages {
+    final public static String MESSAGE_COMMAND_ADD = "add an item into your list.";
+    final public static String MESSAGE_COMMAND_DELETE = "delete an item from your list.";
+    final public static String MESSAGE_COMMAND_VIEW = "view all items or view an individual items";
```
Comment (3m0W33D) [inline src/main/java/terminus/common/Messages.java:6]: Would need to update modifier for standards,
`public static final`

**Review by 3m0W33D — COMMENTED** [review]
Well done!

**Review by kelvneo — APPROVED** [review]
LGTM

**Review by 3m0W33D — APPROVED** [review]
LGTM

**Pull request merged**